crates/pixi_git/src/resolver.rs

```diff
@@ -81,7 +81,11 @@ impl GitResolver {
         write_guard.begin().await?;
 
         // Fetch the Git repository.
-        let source = GitSource::new(url.as_ref().clone(), client, cache);
+        let source = if let Some(reporter) = reporter {
+            GitSource::new(url.as_ref().clone(), client, cache).with_reporter(reporter)
+        } else {
+            GitSource::new(url.as_ref().clone(), client, cache)
+        };
```

Comment:
Maybe we can write this as:

```suggestion
        let source =GitSource::new(url.as_ref().clone(), client, cache);
        let source = if let Some(reporter) = reporter {
            source.with_reporter(reporter)
        } else {
            source
        };
```

To not repeat code.